### PR TITLE
docs(uto): command to remove finalizers from topics

### PR DESCRIPTION
**Pull request by PaulRMellor**
**Documentation**

Adds a *Removing finalizers* section to the UTO content related to deleting managed topics.
Clarifies the reasons for removing finalizers (with caution) and shows the command to remove finalizers. 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/modules/operators/con-deleting-managed-topics.adoc [hunk 1]
```diff
@@ -56,4 +56,31 @@ When the finalization tasks are successfully executed, the finalizer is removed
...
 If the unidirectional Topic Operator is not running, it won't be able to remove the `metadata.finalizer`. 
-Consequently, an attempt to delete the namespace that contains the `KafkaTopic` resource won't complete until either the operator is restarted, or the finalizer is otherwise removed (for example using `kubectl edit`). 
\ No newline at end of file
+Consequently, an attempt to delete the namespace that contains the `KafkaTopic` resource won't complete until either the operator is restarted, or the finalizer is otherwise removed (for example using `kubectl edit`). 
+
+== Removing finalizers
+
+If you want to bypass the finalization process when deleting topics, you can remove the finalizers.   
+You might want to do this to delete topics more quickly; or to handle certain topics without invoking cleanup operations.
+Perhaps the cleanup operation associated with the finalization process is not completing as expected.
```
Comment (scholzj) [inline documentation/modules/operators/con-deleting-managed-topics.adoc:65]: I would probably say something like this:

```suggestion
In order to delete the `KafkaTopic` resource while the Topic Operator is not running, you have to remove the finalizers first.
You can do it manually by editing the resources.
Or you can use one of the example commands below.
```

I think we should stick with the _when the TO is not running_ and not try to explain the situations when that might happen as there might be many different situations.

File: documentation/modules/operators/con-deleting-managed-topics.adoc [hunk 2]
```diff
@@ -56,4 +56,31 @@ When the finalization tasks are successfully executed, the finalizer is removed
 
 Finalizers also prevent related resources from being deleted. 
 If the unidirectional Topic Operator is not running, it won't be able to remove the `metadata.finalizer`. 
-Consequently, an attempt to delete the namespace that contains the `KafkaTopic` resource won't complete until either the operator is restarted, or the finalizer is otherwise removed (for example using `kubectl edit`). 
\ No newline at end of file
+Consequently, an attempt to delete the namespace that contains the `KafkaTopic` resource won't complete until either the operator is restarted, or the finalizer is otherwise removed (for example using `kubectl edit`).
```
Comment (scholzj) [inline documentation/modules/operators/con-deleting-managed-topics.adoc:59]:
```suggestion
And any attempt to delete the `KafkaTopic` resources either directly or for example through namespace deletion will fail or timeout.
```

File: documentation/modules/operators/con-deleting-managed-topics.adoc [hunk 3]
```diff
@@ -56,4 +56,31 @@ When the finalization tasks are successfully executed, the finalizer is removed
 
 Finalizers also prevent related resources from being deleted. 
 If the unidirectional Topic Operator is not running, it won't be able to remove the `metadata.finalizer`.
```
Comment (tombentley) [inline documentation/modules/operators/con-deleting-managed-topics.adoc:58]: The JSON/YAML uses `finalizers` (plural), so maybe reword like this?

```suggestion
If the unidirectional Topic Operator is not running, it won't be able to remove its finalizer from the `metadata.finalizers`. 
```

File: documentation/modules/operators/con-deleting-managed-topics.adoc (hunk 1 above)
Comment (tombentley) [reply documentation/modules/operators/con-deleting-managed-topics.adoc:65]: Perhaps rather than `you have to remove the finalizers first.` we should reword it like this: `you have to remove the finalizers yourself`, because IIRC it's fine to delete the namespace first, then delete the finalizer (i.e. 'first' is incorrect).

File: documentation/modules/operators/con-deleting-managed-topics.adoc [hunk 4]
```diff
@@ -56,4 +56,31 @@ When the finalization tasks are successfully executed, the finalizer is removed
...
+The command uses the `jq` tool to modify the `KafkaTopic` (`kt`) resources by setting the finalizers to `null`.
+You can also use the command for a specific topic:
+
+.Removing a finalizer on a specific topic
+[source,shell]
+----
+kubectl get kt <topic_name> -o=json | jq '.metadata.finalizers = null' | kubectl apply -f -
+----
+
+After running the command, you can go ahead and delete the topics.
```
Comment (tombentley) [inline documentation/modules/operators/con-deleting-managed-topics.adoc:84]:
```suggestion
After running the command, you can go ahead and delete the topics. 
Alternatively, if the topics were already being deleted but were blocked due to outstanding finalizers then their deletion should complete.
```

File: documentation/modules/operators/con-deleting-managed-topics.adoc [hunk 5]
```diff
@@ -56,4 +56,31 @@ When the finalization tasks are successfully executed, the finalizer is removed
...
+
+WARNING: Be careful when removing finalizers, as any cleanup operations associated with the finalization process are not performed.
```
Comment (tombentley) [inline documentation/modules/operators/con-deleting-managed-topics.adoc:86]: Can we make this more specific?

Comment (PaulRMellor) [reply documentation/modules/operators/con-deleting-managed-topics.adoc:85]: Thanks @tombentley -- is this the kind of thing:

WARNING: Be careful when removing finalizers, as any cleanup operations associated with the finalization process are not performed. 
For example, if you remove the finalizer from a `KafkaTopic` resource and subsequently delete the resource, the related topic in Kafka won't be deleted.

**Review by scholzj — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

File: documentation/modules/operators/con-deleting-managed-topics.adoc [hunk 6]
```diff
@@ -7,7 +7,7 @@
 
 [role="_abstract"]
 Unidirectional topic management supports the deletion of topics managed through the `KafkaTopic` resource with or without Kubernetes finalizers.
```
Comment (fvaleri) [inline documentation/modules/operators/con-deleting-managed-topics.adoc:9]: In the YAML examples below, the finalizer is a list element, so it should be:

```sh
  finalizers:
    - strimzi.io/topic-operator
```

File: documentation/modules/operators/con-deleting-managed-topics.adoc [hunk 7]
```diff
@@ -54,6 +54,34 @@ If the deletion fails, it is shown in the status of the resource.
...
-If the unidirectional Topic Operator is not running, it won't be able to remove the `metadata.finalizer`. 
-Consequently, an attempt to delete the namespace that contains the `KafkaTopic` resource won't complete until either the operator is restarted, or the finalizer is otherwise removed (for example using `kubectl edit`). 
\ No newline at end of file
+Finalizers also serve to prevent related resources from being deleted. 
+If the unidirectional Topic Operator is not running, it won't be able to remove its finalizer from the `metadata.finalizers`. 
+And any attempt to delete the `KafkaTopic` resources directly or by deleting the namespace will result in failure or timeout. 
+
+== Removing finalizers
+
+If you want to bypass the finalization process when deleting topics, you have to remove the finalizers.
```
Comment (fvaleri) [inline documentation/modules/operators/con-deleting-managed-topics.adoc:63]:
```suggestion
If the Topic Operator is not running, and you want to bypass the finalization process when deleting topics, you have to remove the finalizers.
```

File: documentation/modules/operators/con-deleting-managed-topics.adoc [hunk 8]
```diff
@@ -54,6 +54,34 @@ If the deletion fails, it is shown in the status of the resource.
 
 When the finalization tasks are successfully executed, the finalizer is removed from the metadata, and the resource is fully deleted.  
 
-Finalizers also prevent related resources from being deleted. 
-If the unidirectional Topic Operator is not running, it won't be able to remove the `metadata.finalizer`. 
-Consequently, an attempt to delete the namespace that contains the `KafkaTopic` resource won't complete until either the operator is restarted, or the finalizer is otherwise removed (for example using `kubectl edit`). 
\ No newline at end of file
+Finalizers also serve to prevent related resources from being deleted. 
+If the unidirectional Topic Operator is not running, it won't be able to remove its finalizer from the `metadata.finalizers`. 
+And any attempt to delete the `KafkaTopic` resources directly or by deleting the namespace will result in failure or timeout.
```
Comment (fvaleri) [inline documentation/modules/operators/con-deleting-managed-topics.adoc:59]: I think the namespace deletion being "stuck in a terminating state" will be a common pitfall. Do we want to mention it more clearly?

**Comment by scholzj** [comment]
@fvaleri @tombentley Do you have something more to this?

**Review by fvaleri — APPROVED** [review]

**Review by tombentley — APPROVED** [review]
LGTM, thanks Paul!
